spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerObjectMapperImpl.java

```diff
@@ -0,0 +1,183 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core.mapping;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+import com.google.cloud.spanner.Mutation.WriteBuilder;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Struct;
+import com.google.cloud.spanner.Type;
+import com.google.cloud.spanner.ValueBinder;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.data.mapping.PersistentPropertyAccessor;
+import org.springframework.data.mapping.PropertyHandler;
+import org.springframework.util.Assert;
+
+/**
+ * @author Chengyuan Zhao
+ */
+public class SpannerObjectMapperImpl implements SpannerObjectMapper {
+
+	private final SpannerMappingContext spannerMappingContext;
+
+	public SpannerObjectMapperImpl(SpannerMappingContext spannerMappingContext) {
+		Assert.notNull(spannerMappingContext,
+				"A valid mapping context for Spanner is required.");
+		this.spannerMappingContext = spannerMappingContext;
+	}
+
+	@Override
+	public <T> List<T> mapToUnmodifiableList(ResultSet resultSet, Class<T> entityClass) {
+		ArrayList<T> result = new ArrayList<>();
+		while (resultSet.next()) {
+			result.add(read(entityClass, resultSet.getCurrentRowAsStruct()));
+		}
+		return Collections.unmodifiableList(result);
+	}
+
+	@Override
+	public <R> R read(Class<R> type, Struct source) {
+		R object = BeanUtils.instantiateClass(type);
```

Comment:
why are we using a BeanUtils instantiation? A simple no-arg constructor can be assumed for entities.

Reply:
will change to use standard Java object creation

Reply:
Check what happens if the columns in struct don't match what's in entity and vice versa for write.